Source/Models/BrickDimension.swift

```diff
@@ -18,6 +18,41 @@ public struct BrickSize {
     }
 }
 
+public enum RestrictedSize {
+    case minimumSize(size: BrickDimension)
+    case maximumSize(size: BrickDimension)
+}
+
+public struct RestrictedBrickSize {
+    let size: BrickDimension
+    let restrictedSize: RestrictedSize
+    
+    var isEstimate: Bool {
+        switch size {
```

Comment:
The switch should be done on `size.dimension(withValue: value)`, because this could be an indirect size